pkg/policy/snapshots.go

```diff
@@ -0,0 +1,204 @@
+package policy
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/jeremywohl/flatten"
+)
+
+func (ce *Executor) StoreSnapshot(ctx context.Context, path string, tables []string) error {
+	if len(tables) == 0 {
+		return errors.New("no tables to snapshot")
+	}
+
+	for _, table := range tables {
+		ef, err := os.OpenFile(fmt.Sprintf("%s/table_%s.csv", path, table), os.O_CREATE|os.O_WRONLY, 0777)
+		if err != nil {
+			fmt.Println("error opening file:", err)
+			return err
+		}
+		ce.log.Info("exporting", "table", table)
+
+		err = ce.conn.RawCopyTo(ctx, ef, fmt.Sprintf("COPY (select * from %s) TO STDOUT DELIMITER '|' CSV HEADER", table))
+		if err != nil {
+			return fmt.Errorf("error exporting file: %+v", err)
+		}
+		ef.Close()
```

Comment:
Close on RawCopyTo err as well? maybe wrap in a `func() error` so Close can be deferred, which would also close on panic etc?